### PR TITLE
feat: Implemented EnsureVoMember filter

### PR DESCRIPTION
**Pull request by BaranekD**

### CODE REVIEW

**Comment by vyskocilpavel** [comment]
Waiting for the synchronization with the OIDC Implementation

File: lib/Auth/Process/EnsureVoMember.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            return;
+        }
+
+        if (empty($attrValues[$this->voDefsAttr])) {
+            Logger::debug(
+                'perun:EnsureVoMember: skip execution - attribute ' . self::VO_DEFS_ATTR . ' has null or no value'
+            );
+            return;
+        }
+        $voShortName = $attrValues[$this->voDefsAttr];
```
Comment (dBucik) [inline lib/Auth/Process/EnsureVoMember.php:111]: to stay consistent with the code above, I would first do the assignment to the variable and then use it in the If statement :)

File: lib/Auth/Process/EnsureVoMember.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            HTTP::redirectTrustedURL(
+                $url,
+                [
+                    'StateId' => $id,
+                    'administrationContact' => $request['SPMetadata']['administrationContact'],
+                    'serviceName' => $request['SPMetadata']['name']['en']
+                ]
+            );
+        }
+    }
```
Comment (dBucik) [inline lib/Auth/Process/EnsureVoMember.php:169]: I don't like this construct. I would prefer if the parameters building would be the first block of code, like:
```
$params = [];
$params['StateId'] = $id;
$params['administrationContact'] = $request['SPMetadata']['administrationContact'];
$params['serviceName'] = $request['SPMetadata']['name']['en'];
if (isset($request['SPMetadata']['InformationURL']['en'])) {
    $params['informationURL'] = $request['SPMetadata']['InformationURL']['en'];
}
HTTP::redirectTrustedURL($url, $params);
```

It's a lot shorter :)

File: lib/Auth/Process/EnsureVoMember.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            !empty($this->registrarUrl) &&
+            $this->adapter->hasRegistrationFormByVoShortName($voShortName)
+        ) {
+            $this->redirectDirectly($loginUrl, $voShortName);
+        } else {
+            $this->redirectUnapproved($request);
+        }
+    }
+
+    private function redirectDirectly($loginUrl, $voShortName)
```
Comment (dBucik) [inline lib/Auth/Process/EnsureVoMember.php:133]: Maybe rename the method to "redirectToRegistration" to better explaing what is the method doing.

**Review by dBucik — CHANGES_REQUESTED** [review]
Some small code improvements (in style, not in functionality), otherwise GTG by me.

**Review by dBucik — APPROVED** [review]

**Review by vyskocilpavel — CHANGES_REQUESTED** [review]
Missing example configuration

**Review by vyskocilpavel — APPROVED** [review]

**Pull request merged**